### PR TITLE
Fix long display name source question breakout

### LINKED ISSUE #30948: [MLv2] [Bug] `long-display-name` should keep the LHS part when coming from a source question
Follow up on #30857

### To Reproduce

1. New > Question > Sample Database > Orders
2. Join the Products table
3. Save the question
4. New > Question > Saved Questions > pick the question you just created
5. Click the "Pick a column to group by" in the summarization section of the editor
6. Click the "Product → Category" column

**Expected:** the clause should be called "Product → Category" as in the source question
**Actual:** the clause is called "Category"

[Related e2e test](https://github.com/metabase/metabase/blob/26b01750e98b328cdd19ca09bb2cb2cc1a51de5e/e2e/test/scenarios/question/nested.cy.spec.js#L272)

### PR DESCRIPTION
**Pull request by metamben**
Fixes #30948

Supersedes #31253.

### CODE REVIEW

File: test/metabase/lib/breakout_test.cljc [hunk 1]
```diff
@@ -439,5 +439,5 @@
                           {:breakout [[:field {:base-type :type/Date, :effective-type :type/Date} "expr"]]}]}
                 query'))
         (testing "description"
-          (is (= "Grouped by Expr"
+          (is (= "Grouped by expr"
```
Comment (metamben) [inline test/metabase/lib/breakout_test.cljc:442]: This seems to be the right result to me, but I might be wrong.

File: src/metabase/automagic_dashboards/core.clj [hunk 2]
```diff
@@ -1054,7 +1054,6 @@
         (apply-rule root (rules/get-rule rule)))
       (some
        (fn [rule]
-         (tap> rule)
```
Comment (metamben) [inline src/metabase/automagic_dashboards/core.clj:1057]: This is a random fix.

File: src/metabase/lib/core.cljc [hunk 3]
```diff
@@ -81,7 +81,8 @@
   [lib.breakout
    breakout
    breakoutable-columns
-   breakouts]
+   breakouts
+   breakouts-metadata]
```
Comment (metamben) [inline src/metabase/lib/core.cljc:85]: Unrelated.

File: src/metabase/lib/field.cljc [hunk 4]
```diff
@@ -78,6 +78,7 @@
         ;; we should look in to fixing this if we can.
         stage-columns (or (:metabase.lib.stage/cached-metadata stage)
                           (get-in stage [:lib/stage-metadata :columns])
+                          (lib.metadata.calculation/default-columns query stage-number stage)
```
Comment (metamben) [inline src/metabase/lib/field.cljc:81]: This is the actual fix. It seems potentially wasteful to calculate the default columns every time we need to resolve a column name.

**Review by camsaul — APPROVED** [review]
